check_diff/tests/git.rs

```diff
@@ -0,0 +1,25 @@
+use check_diff::clone_git_repo;
+
+use std::fs;
+use tempfile::Builder;
+
+#[test]
+fn clone_repo_test() {
+    // Creates an empty directory in the current working directory
+    let dir = Builder::new().tempdir_in("").unwrap();
+    let sample_repo = "https://github.com/rust-lang/rustfmt.git";
+    let dest_path = dir.path();
+    let result = clone_git_repo(sample_repo, dest_path);
+    assert_eq!(result.is_ok(), true);
+    let directory = fs::read_dir(dest_path);
+    // check that the directory is non-empty
+    assert_eq!(directory.iter().next().is_none(), false);
```

Comment:
Probably better to do a positive assertion. Also, you can use `.is_some()` along with `assert!` to check the boolean.

```suggestion
    let directory = fs::read_dir(dest_path);
    // check that the directory is non-empty
    assert!(directory.iter().next().is_some());
```

Reply:
This check is a little unnecessary since we're about to check that a `.git` folder exists, which would also prove that the directory isn't empty.